src/GitVersionCore/Extensions/StringExtensions.cs

```diff
@@ -132,7 +132,7 @@ public static string RegexReplace(this string input, string pattern, string repl
 
         public static bool IsEquivalentTo(this string self, string other)
```

Comment:
It would be great if we could delete `IsEquivalentTo` and rather use the different `.Equals()` implementations to replicate the behavior. But I'm not sure we need to compare `string` as often as we do, or at all.

Reply:
`IsEquivalentTo` is used in several places, including comparing branch names, but not only. So we keep it for now